sdk/communication/chat/src/main/java/com/azure/android/communication/chat/implementation/AzureCommunicationChatServiceImpl.java

```diff
@@ -0,0 +1,2309 @@
+import com.azure.android.core.http.ServiceClient;
+import com.azure.android.core.http.exception.HttpResponseException;
+import com.azure.android.core.http.responsepaging.AsyncPagedDataCollection;
+import com.azure.android.core.http.responsepaging.AsyncPagedDataRetriever;
+import com.azure.android.core.http.responsepaging.PagedDataResponseCollection;
+import com.azure.android.core.http.responsepaging.PagedDataResponseRetriever;
+import com.azure.android.core.internal.util.serializer.SerializerAdapter;
+import com.azure.android.core.internal.util.serializer.SerializerFormat;
+import com.azure.android.core.util.paging.Page;
+import com.azure.android.core.util.paging.PagedDataCollection;
+import com.azure.android.core.util.paging.PagedDataRetriever;
+import okhttp3.RequestBody;
+import okhttp3.ResponseBody;
+import org.threeten.bp.OffsetDateTime;
+import retrofit2.Call;
+import retrofit2.http.Body;
+import retrofit2.http.DELETE;
+import retrofit2.http.GET;
+import retrofit2.http.PATCH;
+import retrofit2.http.Path;
+import retrofit2.http.POST;
+import retrofit2.http.Query;
+
+/**
+ * Initializes a new instance of the AzureCommunicationChatService type.
+ */
+public final class AzureCommunicationChatServiceImpl {
+    /**
+     * The proxy service used to perform REST calls.
+     */
+    private final AzureCommunicationChatServiceService service;
+
+    /**
+     * The serializer.
+     */
+    final SerializerAdapter serializerAdapter = SerializerAdapter.createDefault();
+
+    /**
+     * The Azure Core generic ServiceClient to setup interceptors and produce retrofit proxy.
+     */
+    private ServiceClient serviceClient;
+
+    /**
+     * Gets The Azure Core generic ServiceClient to setup interceptors and produce retrofit proxy.
+     * 
+     * @return the serviceClient value.
+     */
+    public ServiceClient getServiceClient() {
+        return this.serviceClient;
+    }
+
+    /**
+     * The endpoint of the Azure Communication resource.
+     */
+    private final String endpoint;
+
+    /**
+     * Gets The endpoint of the Azure Communication resource.
+     * 
+     * @return the endpoint value.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+    /**
+     * Api Version.
+     */
+    private final String apiVersion;
+
+    /**
+     * Gets Api Version.
+     * 
+     * @return the apiVersion value.
+     */
+    public String getApiVersion() {
+        return this.apiVersion;
+    }
+
+    /**
+     * Initializes an instance of AzureCommunicationChatService client.
+     * 
+     * @param serviceClient The Azure Core generic ServiceClient to setup interceptors and produce retrofit proxy.
+     * @param endpoint The endpoint of the Azure Communication resource.
+     */
+    public AzureCommunicationChatServiceImpl(ServiceClient serviceClient, String endpoint) {
+        this.serviceClient = serviceClient;
+        this.endpoint = endpoint;
+        this.apiVersion = "2020-09-21-preview2";
```

Comment:
In the future we might want to provide passing `ServiceVersions` types to our clients (e.g. `ChatServiceVersion`) type to allow for supporting multiple service versions, just like Java SDKs do.

Reply:
Make sense. I think in JavaSDK, it's in the convenience layer. The first release will be AutoGen and future release there is plans to add a convenience layer.